src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -218,43 +403,58 @@ void extractRecurrence(const QJsonArray &recurrence, KCalCore::Event::Ptr event,
         return;
     }
 
+    kcalRecurrence->clear(); // avoid adding duplicate recurrence information
```

Comment:
By recurs() check you mean the size()? It could cover but unsetRecurs() is specified not to clear explicit dates, only rules. clear() is specified to remove all so in this current version unsetRecurs() is redundant. And anyway, if clear() is called here before inserting any new recurrences, the whole block looks unnecessary to me, for loop below is no-op with no recurrences.